o-fish-ios/Helpers/UIImage+Thumbnail.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  UIImage+Thumbnail.swift
+//  
+//  Created on 6/16/20.
+//  Copyright © 2020 WildAid. All rights reserved.
+//
+
+import UIKit
+
+extension UIImage {
+    func thumbnail(size: CGFloat) -> UIImage? {
+        var thumbnail: UIImage?
+        if let imageData = self.pngData() {
+            let options = [
+                kCGImageSourceCreateThumbnailWithTransform: true,
+                kCGImageSourceCreateThumbnailFromImageAlways: true,
+                kCGImageSourceThumbnailMaxPixelSize: size] as CFDictionary
+
+            imageData.withUnsafeBytes { ptr in
```

Comment:
check is this call sync, cause function will return nil if it will finish earlier

Reply:
I used debugging for checking. Function return after execution closure.